project-chip/connectedhomeip
# [TE4] Add ping command to python device controller for MRP testing

#### Problem
What is being fixed? 
* Currently CSG use chip_shell and chip-echo-responder to test secure channel, and chip-echo-responder has only been enabled on selected platforms with testSecurePairingSecret. we have requirements from CSG to validate secure channel using example apps on real devices over PASE/CASE. We need to find a way to test secure channel and MRP against real example app over PASE/CASE. 
  
* Fixes #6257 Echo Sever does't work on the applications with chip-shell 

#### Change overview
Add ping command to python device controller

#### Testing
How was this tested? (at least one bullet point required)

1. Build chip-lighting-app with echo enabled 
`gn gen out/debug --args='chip_app_use_echo=true'`
2. Start ./chip-lighting-app on the DUT.
3. Run chip-device-ctrl from controller side.
3. Confirm 'ping' command is enabled
```
chip-device-ctrl > ping ?
Usage:

ping [options] <nodeid>

Options:
  -i  <interval>  ping interval time in seconds
  -c  <count>    stop after <count> replies
  -r                   enable MRP
  -s  <size>      payload size in bytes

ping command is using Echo Protocol to measure packet loss across network paths.
```

4. Establish secure connection with device under test
`connect -ip <address> <SetUpPinCode> [<nodeid>]`

5. Ping DUT from python controller
`chip-device-ctrl > ping 12344321`

6. Confirm all echo request have been echoed form DUT
`[1623093477.920869][2573955] CHIP:CTL: Echo Response: 3/3(100.00%) len=32 time=376959.904ms`
